packages/assets-controllers/src/TokenDetectionController.test.ts

```diff
@@ -524,73 +486,44 @@ describe('TokenDetectionController', () => {
     const onPreferencesStateChange = sinon.stub().callsFake((listener) => {
       preferencesStateChangeListener = listener;
     });
-    tokenDetection = new TokenDetectionController(
-      {
-        onPreferencesStateChange,
-        onTokenListStateChange: stub,
-        onNetworkStateChange: stub,
-        getBalancesInSingleCall: getBalancesInSingleCallMock,
-        addDetectedTokens: stub,
-        getTokensState: () => tokensController.state,
-        getTokenListState: () => tokenList.state,
-        getNetworkState: () => defaultNetworkState,
-        getPreferencesState: () => preferences.state,
-        getNetworkClientById: jest.fn(),
-      },
-      {
-        disabled: false,
-        isDetectionEnabledForNetwork: true,
-        isDetectionEnabledFromPreferences: false,
-        selectedAddress: '0x1',
-      },
-    );
+    tokenDetection = new TokenDetectionController({
+      chainId: ChainId.mainnet,
+      selectedAddress: '0x1',
+      onPreferencesStateChange,
+      getBalancesInSingleCall: getBalancesInSingleCallMock,
+      addDetectedTokens: stub,
+      getTokensState: () => tokensController.state,
+      getPreferencesState: () => preferences.state,
+      messenger: buildTokenDetectionControllerMessenger(controllerMessenger),
+    });
 
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     await preferencesStateChangeListener!({
       selectedAddress: '0x1',
       useTokenDetection: true,
     });
-
+    await tokenDetection.start();
```

Comment:
`tokenDetection` here was previously initialized with a `disabled: false` option, which was why the `.start()` call wasn't necessary. However, I removed the `disabled` class field since it's a property of `BaseConfig`, which is no longer being used in `BaseControllerV2`. 

I ended up replacing the `disabled` option with `deferPollingStart: false`, which removes the need for the `.start()` call:
https://github.com/MetaMask/core/pull/3609/commits/fdb70d9230013b9c6b46cfb666439b82a51dc1b2

Reply:
Restored here https://github.com/MetaMask/core/pull/3609/commits/3532bbc275187e36dcd2aea5cd2aa98479fd3ab6 based on our discussion on removing `deferPollingStart`.